src/emhass/web_server.py

```diff
@@ -49,46 +102,64 @@ def action_call(action_name):
     input_data_dict = set_input_data_dict(config_path, str(data_path), costfun, 
```

Comment:
There are a few places errors can occur if url or key is wrong, these lines are one. (then the other two are on ML Fit & Publish)

A suggested change for this line could be:
```python
    ActionStr = " >> Test HA connection..."
    app.logger.info(ActionStr)
    try:
        input_data_dict = set_input_data_dict(config_path, str(data_path), costfun, 
            params, runtimeparams, action_name, app.logger)
    except Exception as e :
        return make_response(grabLog(ActionStr), 400)
```

Reply:
The other method you could do is rap each action in a try and expect. Expect pasting exception in app.logger.error then returning make_response